content/francais/python-turtle/activity-4.md

```diff
@@ -0,0 +1,77 @@
+---
+title: "Activité 4 - Simplifier le code avec des fonctions"
+date: 2024-09-17T13:24:17-07:00
+weight: 6
+draft: false
+---
+
+Nous avons dessiné un hexagone. Mais ce n'est qu'un des nombreux hexagones qu'Alex doit construire pour sa ruche. Encore une fois, il semble que nous devrions répéter le code plusieurs fois pour dessiner plusieurs hexagones. Heureusement, nous n'en avons pas besoin ! Les fonctions à la rescousse !
+
+{{% notice tip %}}
+
+## Fonctions
+
+Une fonction est un moyen de regrouper des lignes de code pour faire quelque chose. Par exemple, `turtle.forward(50)` déplace la tortue de 50 pas en avant, et `turtle.left(120)` tourne la tortue de 120 degrés à gauche. Que faire si nous voulons toujours effectuer ces deux actions ensemble ? Nous pouvons les mettre dans une fonction appelée `draw_line()`, de sorte que chaque fois que cette fonction est appelée, ces deux actions se produisent simultanément. Pensez-y comme une recette : nous regroupons toutes les instructions ensemble, et le nom de la recette est la nourriture que nous préparons.
+
+{{% /notice %}}
+
+Voici un exemple de **définition de fonction**, et comment nous l'utiliserions :
+
+``` python
+def draw_line():
+  turtle.forward(50)
+  turtle.left(120)
+```
+
+La première ligne est appelée l'**en-tête** de définition de fonction. Le mot-clé `def` indique à l'ordinateur que nous définissons une nouvelle fonction. Ensuite, nous donnons un nom à la fonction, dans ce cas `draw_line`. Enfin, n'oubliez pas le `():` à la fin de la ligne, qui indique à l'ordinateur que nous commençons le corps de la fonction.
+
+{{% notice tip %}}
+
+## Matériel supplémentaire - paramètres
+
+Dans les parenthèses `()` nous définirions normalement des **paramètres** pour la fonction. Les paramètres sont des entrées que nous pouvons donner à la fonction, que la fonction peut utiliser pour produire sa sortie. Par exemple, nous devons avoir deux nombres pour les additionner et produire un résultat - les deux nombres que nous utilisons peuvent être considérés comme des **paramètres**. Aujourd'hui, nous n'utiliserons pas de paramètres, et nous laisserons comme `()`.
+
+{{% /notice %}}
+
+Semblable à une boucle for, tout ce que nous souhaitons définir dans la fonction doit être précédé d'une tabulation.
+
+Si vous appuyez sur **run** avec juste ce code, vous remarquerez que vous ne verrez aucune sortie ! Nous avons seulement créé les fonctions, mais nous devons les utiliser. Pour utiliser la fonction que nous avons créée, tapez à nouveau `draw_line()` sans le `def`.
+
+``` python
+def draw_line():    # Définition de fonction
+  turtle.forward(50)
+  turtle.left(120)
+
+draw_line()         # Appel de fonction
+```
+
+Ceci est appelé un **appel de fonction** pour la fonction `draw_line()`. Un appel de fonction exécute le code défini dans la fonction portant le même nom. Avec l'analogie de la recette, pensez à cela comme à la réalisation de la recette.
```

Comment:
This is correctly translated: "Ceci est appelé un **appel de fonction** "
however, I see in line 38 that it does not translated the items inside the asterisks, but in line 48 it did. To keep in mind in case you are specifying something in the prompt.